pkg/operator/controller/controller_cluster_role.go

```diff
@@ -15,37 +15,37 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 )
 
-func (r *reconciler) ensureDNSClusterRole() (*rbacv1.ClusterRole, error) {
-	current, err := r.currentDNSClusterRole()
+func (r *reconciler) ensureDNSClusterRole() (bool, *rbacv1.ClusterRole, error) {
+	haveCR, current, err := r.currentDNSClusterRole()
 	if err != nil {
-		return nil, err
+		return false, nil, err
 	}
 	desired := desiredDNSClusterRole()
 
 	switch {
-	case desired != nil && current == nil:
+	case !haveCR:
 		if err := r.client.Create(context.TODO(), desired); err != nil {
-			return nil, fmt.Errorf("failed to create dns cluster role: %v", err)
+			return false, nil, fmt.Errorf("failed to create dns cluster role: %v", err)
 		}
 		logrus.Infof("created dns cluster role: %s/%s", desired.Namespace, desired.Name)
-	case desired != nil && current != nil:
+	case haveCR:
 		if err := r.updateDNSClusterRole(current, desired); err != nil {
```

Comment:
Sidenote:  A future cleanup could add a Boolean return value to the update methods that don't have one to indicate whether an update was performed.  If no update was performed, the caller can return `true, current, nil` instead of falling through to `return r.currentFoo` (which incurs an unnecessary API call).

Reply:
Good idea! I suspect there are quite a few places that have these extra API calls, not just in the cluster-dns-operator but also in the cluster-ingress-operator, right?

Reply:
Probably, yeah.  This is an opportunity to make all of this logic more consistent, correct, and efficient across both operators.